content/tutorials/7.workflows/invincible-ai-content-workflows-with-inngest-and-directus.md

```diff
@@ -0,0 +1,608 @@
+
+Most importantly, Inngest eliminates the need to manage complex queue infrastructure. You simply write functions in your existing codebase using Inngest's SDK, and it handles all the orchestration for you. This approach is particularly valuable with AI workflows, which often require careful state management and can benefit from the step-by-step execution model.
+
+**Here's a quick fictional example of an Inngest function.**
+
+```tsx
+// Example of an Inngest function with steps
+export const analyzeContent = inngest.createFunction(
+  { id: "analyze-content" },
+  { event: "content/created" },
+  async ({ event, step }) => {
+    // Steps are atomic, durable operations that retry on failure
+    const extractedText = await step.run("extract-text", async () => {
+      return textExtractor.process(event.data.documentUrl)
+    })
+
+    // State is automatically preserved between steps
+    const analysis = await step.run("analyze-with-ai", async () => {
+      return aiService.analyze(extractedText)
+    })
+
+    // Final results can be saved or further processed
+    return analysis
+  }
+)
+
+```
+
+The platform also offers a developer-friendly experience with excellent local development tools, comprehensive observability, and tools for debugging and recovery when things go wrong in production.
+
+## The Directus + Inngest Integration: Beyond Flows
+
+While Directus already includes its own [workflow automation system (Flows)](/guides/automate/flows), Inngest complements it by handling scenarios that Flows wasn't designed for. Directus Flows excels at short-lived automations like sending notifications or processing simple data operations, but AI workflows typically require more resilience and computational power.
+
+Inngest is the perfect companion when you need:
+
+- **Long-running processes** that might take minutes or hours (versus Flows' seconds)
+- **Complex, multi-step AI workflows** with dependencies and state management
+- **Compute-intensive tasks** that shouldn't block your main application
+- **Sophisticated error handling** with automatic retries
+- **Resource isolation** to keep your Directus instance responsive
+
+By integrating Directus with Inngest, you create a content management system capable of sophisticated AI operations like content translation, image analysis, or generative AI tasks while maintaining performance. You can even trigger Inngest functions from Directus Flows, combining the visual simplicity of Flows with the computational power of Inngest for heavy processing.
+
+Now, let's dive into the technical implementation.
+
+---
+
+## Implementation Guide
+
+
+::callout{icon="material-symbols:info-outline"}
+
+To implement this integration, you'll want to have
+
+- A basic understanding of [Directus Extensions](/guides/extensions/overview).
+
+::
+
+
+Let's walk through the complete implementation process step by step, following a logical development workflow.
+
+### 1. Set Up Your Directus Environment
```

Comment:
What do you think about removing the numbers from the steps? They don't feel right for sub-headings

Reply:
Thank you! Let's do it for consistency, please